.github/workflows/build-special.yml

```diff
@@ -0,0 +1,153 @@
+name: Special Cases CI
+
+on: [pull_request]
+
+jobs:
+  build:
+    name: ${{ matrix.name }} ${{ matrix.build_type }}
+    runs-on: ${{ matrix.os }}
+
+    env:
+      CTEST_OUTPUT_ON_FAILURE: ON
+      CTEST_PARALLEL_LEVEL: 2
+      CMAKE_BUILD_TYPE: ${{ matrix.build_type }}
+      GTSAM_BUILD_UNSTABLE: ON
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        # Github Actions requires a single row to be added to the build matrix.
+        # See https://help.github.com/en/articles/workflow-syntax-for-github-actions.
+        name: [
+            ubuntu-gcc-deprecated,
+            ubuntu-gcc-quaternions,
+            ubuntu-gcc-points-vector,
+            macos-xcode-deprecated,
+            macos-xcode-quaternions,
```

Comment:
I think we just need to test all these with gcc on ubuntu, to reduce CI estate.

Reply:
Sounds good. I wasn't sure what coverage we were aiming for hence I threw the kitchen sink at it.